### DISCUSSION TITLE
Problema Google Cast

### DISCUSSION THREAD

**Discussion by dmalta87** [indomus/forum#1334] [HUB: Home Assistant]
Buonasera a tutti, mi chiamo Dario e mi sono avvicinato da pochissimo all'esperienza Home Assistant.
Ho configurato svariate integrazioni, componenti aggiuntivi e quant'altro, seguendo le guide trovate sul web.
Ho un problema, a cui non riesco a venire a capo da giorni, con Google Cast: Home Assistant riconosce i miei 2 smart speaker (Google Home Mini e Google Nest Hub 2), mi propone l'integrazione, mi crea i dispositivi, ma mi fa vedere le entità "non disponibili" con un cerchio rosso con punto esclamativo bianco. Di fatto non riesco ad utilizzare i 2 smart speaker per riprodurre nulla. Non so se centra qualcosa, ma ho configurato Duck DNS e NGINX per l'accesso da remoto ed ho integrato Google Home / Google Assistant con i servizi Google Cloud Platform (GCP).
Grazie anticipo a chi proverà ad aiutarmi.

**Comment by snakuzzo** [comment]
In realtà non dovresti vederli con quel pallino.
Ad ogni modo per quei device dovresti impostare nelle impostazioni generali (da GUI o via YAML) `internal_url` ed `external_url` con le rispettive URL di accesso ad HA.

Es.

```yaml
homeassistant:
  internal_url: http://192.168.1.10:8123
  external_url: https://tuodominio.duckdns.org:9000
```

**Reply by dmalta87** [reply]
grazie per il supporto... ma non cambia nulla, continuo a non vederl
![Schermata](https://user-images.githubusercontent.com/97636470/149748175-1596919e-c1ca-46fc-91a4-569f639c3cd5.png)
i

**Reply by snakuzzo** [reply]
Ti conviene controllare il log di HA.
Ad ogni modo le tre entità con il solo bordino rosso sono probabilmente entità che avevi su HA e che ora non ci sono più...e puoi rimuoverle (anche da GUI)

**Reply by dmalta87** [reply]
Questo è quanto...

2022-01-17 12:25:31 ERROR (Thread-6) [pychromecast.socket_client] [TV Cameretta(192.168.1.51):8009] Failed to connect to service ServiceInfo(type='mdns', data='MiTV-MSSP2-338d3595967ba1d7c8dc95192c20e239._googlecast._tcp.local.'), retrying in 5.0s

2022-01-17 12:25:31 ERROR (Thread-7) [pychromecast.socket_client] [Sala(192.168.1.104):8009] Failed to connect to service ServiceInfo(type='mdns', data='Google-Nest-Hub-9f015acbfb363f3133b0acf05060d064._googlecast._tcp.local.'), retrying in 5.0s

2022-01-17 12:25:32 ERROR (Thread-8) [pychromecast.socket_client] [Corridoio(192.168.1.100):8009] Failed to connect to service ServiceInfo(type='mdns', data='Google-Home-Mini-dbf6a7cb9bfd55db8dfe99d307625513._googlecast._tcp.local.'), retrying in 5.0s

**Reply by snakuzzo** [reply]
Dai un'occhiata qui. Forse dipende dalla "tua" rete...

https://community.home-assistant.io/t/multiple-devices-with-different-integrations-are-becoming-unavailable/313369

**Reply by dmalta87** [reply]
io ti ringrazio... ma non capisco come risolvere... probabilmente non ne capisco abbastanza... in google home vedo i dispositivi di Home Assistant, e riesco a comandarli con la voce, non riesco invece a fare l'inverso, cioè far sentire musica / audio da Home Assistant usando i google come speaker

**Reply by snakuzzo** [reply]
sono due cose diverse.
Pilotare i device con la voce...il giro è

SPEAKER -> INTERNET -> IP PUBBLICO TUA ISTANZA DI HA

Far parlare lo speaker invece è...

TUA VOCE -> TUA ISTANZA HA -> SPEAKER

Il fatto è che la tua istanza forse non riesce a comunicare correttamente con gli speaker. Per questo, come indicavano nel post, potrebbe essere un problema legato alla TUA rete

**Reply by dmalta87** [reply]
eh, capisco... ma non riesco ad individuare quale potrebbe essere il problema... anche perché in un primo momento, all'installazione di HA avevo attivato la prova del cloud nabu e funzionava tutto... potrebbe essere un problema di porte? di internal url?

**Reply by snakuzzo** [reply]
Certo...quelli vanno impostati.

Nel `configuration.yaml` imposta:

la parte di internal ed external url così (oppure lo fai da lovelace sotto impostazioni > generali)...

```yaml
homeassistant:
  internal_url: http://ipraspberry:8123   # oppure https://ipraspberry:8123 (a seconda se usi oppure no un reverse proxy)
  external_url: http://tuodominio.duckdns.org:9000 
```
...e la prate TTS così...

```yaml
tts:
- platform: google_translate
  service_name: google_say
  language: 'it'
  base_url: https://tuodominio.duckdns.org:9000
```

**Reply by dmalta87** [reply]
Corretto... avevo nelle varie prove fatte, messo l'internal url uguale all'external, col dominio duckdns.
TTS invece non era tutto compilato ma c'era solo la riga "platform".
Risultato? Uguale a prima... non va...

**Reply by snakuzzo** [reply]
Hai un reverse proxy oppure no ?

**Comment by dmalta87** [comment]
NGINX

Dario Maltarolo

**Reply by snakuzzo** [reply]
scusami e allora perché utilizzi la stessa URL interna ed esterna ?

Come URL interna devi mettere l'IP LAN in http e la porta su cui è in ascolto HA

**Comment by dmalta87** [comment]
Difatti era un errore e prima l’ho corretto. Però devo mettere https perché se metto http non si collega.

Dario Maltarolo

**Reply by snakuzzo** [reply]
??? non ho capito che configurazione hai fatto.
Non ha alcun senso mettere HA in https ed il reverse proxy in https...
I certificati li fai gestire al RP che si preoccupa di proxare e girare in http su HA

**Comment by dmalta87** [comment]
Eh, che qualcosa non va è chiaro, ma non capisco dov’è l’errore. Se metto sul browser http://ip:8123 non si apre nulla. Se metto https si

Dario Maltarolo

**Reply by snakuzzo** [reply]
Per favore non creare ogni volta nuovi thread e fai reply direttamente sotto l'ultimo messaggio...e non rispondere direttamente dalle mail :D 
Tornando al tuo problema...dovresti fare in modo di configurare NGINX per la gestione dei certificati ed Home Assistant invece lo lasci in http.

**Reply by dmalta87** [reply]
Ops, scusa. Sono poco pratico.
Questa è la schermata di NGINX… qualche errore qui?
![847BA39C-3956-4BE4-B7B0-19588CE8CBE3](https://user-images.githubusercontent.com/97636470/149833830-a7bc1353-f331-49e1-9c40-0c4da6032d26.jpeg)

**Reply by snakuzzo** [reply]
Non conosco l'addon NGINX.
Ho HassOS solo in test e come Reverse Proxy uso Caddy (abbiamo fatto pure una guida in merito [QUI](https://indomus.it/guide/installare-e-configurare-caddy-su-home-assistant-os-hassos-hassio/) ).
Appena possibile faremo qualche prova anche su NGINX

**Reply by snakuzzo** [reply]
Allora...ho appena rimosso Caddy ed usato il tuo stesso addon.
In base a quello che hai fatto, la configurazione dell'addon mi sembra ok.

Il tuo problema è il `configuration.yaml` di Home Assistant. Probabilmente hai lasciato roba relativa alla configurazione prima dell'utilizzo di NGINX
Come da documentazione ufficiale dell'addon devi portare chiave e certificato sotto `/ssl`, rimuovere quello che hai nel blocco `http` nel `configuration.yaml` di Core e sostituirlo in questo modo...

```yaml
http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 172.30.33.0/24
```

Come vedi è tutto scritto nel tab `Documentation` dell'addo-on

![log](https://user-images.githubusercontent.com/7205371/149835403-62501df0-ff27-4a91-977d-ff76815fe638.png)

Fatto questo...

in locale HA risponderà in HTTP su IP locale (es. `http://192.168.1.10:8123` ... che sarà l' `internal_url`)
da internet invece risponderà in HTTPS su FQDN (es. `https://tuodominio.duckdns.org` ... che sarà l' `external_url` da usare anche come `base_url` nella config del TTS)

...e ti assicuro che è così...perché l'ho appena testato con mano e funziona egregiamente :)

**Reply by dmalta87** [reply]
devo togliere le prime 2 righe?
<img width="279" alt="Schermata 2022-01-17 alle 22 09 50" src="https://user-images.githubusercontent.com/97636470/149837341-af565fa9-d38a-400d-8dfb-8b14693770c2.png">

**Reply by snakuzzo** [reply]
Sì

**Reply by snakuzzo** [reply]
e devi riavviare Core

**Reply by dmalta87** [reply]
fatto e funziona http, ma non va più con duckdns perché mi da errore ssl

**Reply by snakuzzo** [reply]
Leggi la documentazione dell'addon...io non posso sistemarti le cose da qui.
Devi leggere la documentazione e fare quello che c'è scritto.
I certificati li genererai in qualche modo (immagino con l'addon DuckDNS nel tuo caso) e finiranno sotto `/ssl`...ti torna ?
Devi assicurarti che siano esattamente quelli in configurazione dell'addon

**Reply by snakuzzo** [reply]
Assicurati che le porte dell'addon (vedi il tuo screenshot) siano configurate correttamente.
Se hai esposto HA sulla 8123 su internet ... devi configurare l'addon in questo modo...

![log](https://user-images.githubusercontent.com/7205371/149838671-073e29aa-cee9-4e99-a6f5-6ac4b8c76a6c.png)

**Reply by snakuzzo** [reply]
Contattami su Telegram se non risolvi...così magari mi collego da remoto e provo a darti una mano

**Reply by dmalta87** [reply]
Ciao. Giusto per aggiornarti, non cambia nulla con 1 solo modem/router. Penso che il problema sia la mancanza di gestione del mdns del modem.
